src/libostree/ostree-repo-commit.c

```diff
@@ -429,6 +429,25 @@ add_size_index_to_metadata (OstreeRepo        *self,
   return g_variant_ref_sink (g_variant_builder_end (builder));
 }
 
+static gboolean
+throw_min_free_space_error (OstreeRepo  *self,
+                            guint64      bytes_required,
+                            GError     **error)
+{
+  g_autofree char *formatted_required = g_format_size (bytes_required);
+  g_autofree char *err_text = NULL;
+
+  if (bytes_required > 0)
+    err_text = g_strdup_printf ("would be exceeded, at least %s requested", formatted_required);
+  else
+    err_text = g_strdup ("would be exceeded.");
```

Comment:
Cool! Just one final note. There are two ways we could minimize the number of allocations here. First, we can move `formatted_required` inside the `if` branch so we generate it only if we need it. Second, in the `else` case, we don't need to `strdup` since it's a static string. The pattern we usually follow is something like this:

```c
const char *str = NULL;
g_autofree char *str_owned = NULL

if (cond_that_requires_allocation)
  str = str_owned = g_strdup ...
else
  str = "static string"
```

Reply:
Thanks for mentioning the pattern. Addressed your comments.